lighthouse-core/audits/seo/plugins.js

```diff
@@ -0,0 +1,147 @@
+ * @param {string} type
+ */
+function isPluginType(type) {
+  type = type.trim().toLowerCase();
+
+  return TYPE_BLOCKLIST.has(type) ||
+    type.startsWith(JAVA_APPLET_TYPE) || // e.g. "application/x-java-applet;jpi-version=1.4"
+    type.startsWith(JAVA_BEAN_TYPE);
+}
+
+/**
+ * Verifies if given url points to a file that has a known plugin extension
+ * @param {string} url
+ */
+function isPluginURL(url) {
+  try {
+    // in order to support relative URLs we need to provied a base URL
+    const filePath = new URL(url, 'http://example.com').pathname;
+    const parts = filePath.split('.');
+
+    return parts.length > 1 && FILE_EXTENSION_BLOCKLIST.has(parts.pop().trim().toLowerCase());
+  } catch (e) {
+    return false;
+  }
+}
+
+class Plugins extends Audit {
+  /**
+   * @return {!AuditMeta}
+   */
+  static get meta() {
+    return {
+      name: 'plugins',
+      description: 'Document avoids plugins.',
+      failureDescription: 'Document uses plugins',
+      helpText: 'Some types of media or content are not playable on mobile devices. ' +
+      '[Learn more](https://developers.google.com/speed/docs/insights/AvoidPlugins).',
+      requiredArtifacts: ['EmbeddedContent'],
+    };
+  }
+
+  /**
+   * @param {!Artifacts} artifacts
+   * @return {!AuditResult}
+   */
+  static audit(artifacts) {
+    const plugins = artifacts.EmbeddedContent
+      .filter(item => {
+        if (item.tagName === 'APPLET') {
+          return true;
+        }
+
+        if (
+          (item.tagName === 'EMBED' || item.tagName === 'OBJECT') &&
+          item.type &&
+          isPluginType(item.type)
+        ) {
+          return true;
+        }
+
+        const embedSrc = item.src || item.code;
+        if (item.tagName === 'EMBED' && embedSrc && isPluginURL(embedSrc)) {
+          return true;
+        }
+
+        if (item.tagName === 'OBJECT' && item.data && isPluginURL(item.data)) {
+          return true;
+        }
+
+        const failingParams = item.params.filter(param =>
+          SOURCE_PARAMS.has(param.name.trim().toLowerCase()) && isPluginURL(param.value)
+        );
+
+        return failingParams.length > 0;
+      })
+      .map(plugin => {
+        const tagName = plugin.tagName.toLowerCase();
+        const attributes = ['src', 'data', 'code', 'type']
+          .reduce((result, attr) => {
+            if (plugin[attr] !== null) {
+              result += ` ${attr}="${plugin[attr]}"`;
+            }
+            return result;
+          }, '');
+        const params = plugin.params
+          .filter(param => SOURCE_PARAMS.has(param.name.trim().toLowerCase()))
+          .map(param => `<param ${param.name}="${param.value}" />`)
+          .join('');
+
+        return {
+          source: {
+            type: 'node',
+            snippet: `<${tagName}${attributes}>${params}</${tagName}>`,
+          },
+        };
+      });
+
+    const headings = [
+      {key: 'source', itemType: 'code', text: 'Source'},
```

Comment:
Source => Element source

Reply:
Just a note: we are already using 'Source' in `hreflang` and `is-crawlable`